### PR TITLE
[CU-369k8dr] create database connection

### PR DESCRIPTION
**Pull request by vitvara**
Create database connection module

### CODE REVIEW

**Comment by vitvara** [comment]
Task linked: [CU-369k8dr Connect SQL package](https://app.clickup.com/t/369k8dr)

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+MYSQL_URL = "root:<yourMySQLdatabasepassword>@tcp(127.0.0.1:3306)/test"
```
Comment (sirateek) [inline .env:1]: Please do not commit `.env` into repo krub

File: database/databasemysql.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/joho/godotenv"
+)
+
+type Config struct {
+	IP           string
```
Comment (sirateek) [inline database/databasemysql.go:12]: Should we use `Hostname ` instead of IP?

File: database/databasemysql.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	Username     string
+	Password     string
+	DatabaseName string
+}
+
+func CreateConnection(config Config) *sql.DB {
+
+	err := godotenv.Load(".env")
```
Comment (sirateek) [inline database/databasemysql.go:20]: What is this line doing?

File: database/databasemysql.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	// check the connection
+	err = db.Ping()
+
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Success!")
```
Comment (sirateek) [inline database/databasemysql.go:37]: Use logrus instead krub

File: database/databasemysql.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+type Config struct {
+	IP           string
+	Username     string
+	Password     string
+	DatabaseName string
+}
+
+func CreateConnection(config Config) *sql.DB {
```
Comment (sirateek) [inline database/databasemysql.go:18]: I think NewDbConnection is a better name

File: database/databasemysql.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// check the connection
+	err = db.Ping()
+
+	if err != nil {
+		panic(err)
```
Comment (sirateek) [inline database/databasemysql.go:34]: Do not panic. Just return error out.

If so, the code will stop here krub

In go, there is no try except concept. So the only way to check error is to return them to the caller

File: database/databasemysql.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+func CreateConnection(config Config) *sql.DB {
+
+	err := godotenv.Load(".env")
+
+	db, err := sql.Open(
+		"mysql",
+		config.Username+":"+config.Password+"@tcp("+config.IP+")/"+config.DatabaseName,
+	)
+	if err != nil {
+		panic(err)
```
Comment (sirateek) [inline database/databasemysql.go:27]: Do not use panic

**Review by sirateek — APPROVED** [review]

**Pull request merged**